components/director/internal/tenantmapping/handler.go

```diff
@@ -245,16 +247,26 @@ func addTenantsToExtra(objectContexts []ObjectContext, reqData oathkeeper.ReqDat
 }
 
 func addScopesToExtra(objectContexts []ObjectContext, reqData oathkeeper.ReqData) {
-	objScopes := make([][]string, 0, len(objectContexts))
+	objScopes := make([]string, 0)
+scopesBuilder:
 	for _, objCtx := range objectContexts {
-		objScopes = append(objScopes, strings.Split(objCtx.Scopes, " "))
+		currentScopes := strings.Split(objCtx.Scopes, " ")
+		switch objCtx.ScopesMergeStrategy {
+		case overrideAllScopes:
+			objScopes = currentScopes
+			break scopesBuilder
```

Comment:
If a conext with override strategy comes first we will break the loop (`break scopesBuilder`), and not check the other contexts.
If it comes second it will override them, but you've already mentioned that.

The BAS flow should work as expected - it will take the scopes from the token, not the ones from the certificate.
Or am I missing something?

Reply:
Nope, I think it’s okay. I just overlook the loop break.